lib/codegen/src/isa/x86/abi.rs

```diff
@@ -425,6 +425,15 @@ fn insert_common_prologue(
     csrs: &RegisterSet,
     isa: &TargetIsa,
 ) {
+    if stack_size > 0 {
+        // Check if there is a special stack limit parameter. If so insert stack check.
+        if let Some(stack_limit_arg) = pos.func.special_param(ArgumentPurpose::StackLimit) {
+            // TODO: Instead of the `stack_size` we need to pass `stack_frame_size` which
```

Comment:
Reviving the comment about the stack size here:

```
On the second thought, we can't use `stack_size` alone (At least if we want to guarantee that we never modify the stack below the given `StackLimit`). Because it doesn't account for the size required for RBP, all CSRs and potentially stack space needed for `probestack`.
```